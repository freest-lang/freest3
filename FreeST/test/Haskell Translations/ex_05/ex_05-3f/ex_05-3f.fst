-- V exercise 3f

data IntList = End | List Int IntList

filter' : (Int -> Bool) -> IntList -> IntList
filter' f list =
    case list of {
        End -> End,
        List x rest -> 
            if f x 
                then List x (filter' f rest)
                else filter' f rest
    }

biggerThan : Int -> Int -> Bool
biggerThan x y = x < y

list : IntList
list = List 1 (List 2 (List 3 (List 4 (List 5 (List 6 End)))))

main : IntList
main = filter' (biggerThan 5) list
-- result = List 6 End