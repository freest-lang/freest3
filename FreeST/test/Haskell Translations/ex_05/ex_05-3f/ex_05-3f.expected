List 6 End